base/system/winlogon/sas.c

```diff
@@ -1070,6 +1092,9 @@ DoGenericAction(
                     LogOffFlags |= EWX_FORCE;
                 if (!Session->Gina.Functions.WlxIsLogoffOk(Session->Gina.Context))
                     break;
+                /* Prevent LocalSystem from logging off unless it is shutting down. */
+                if (IsLocalSystem(Session->UserToken) && !WLX_SHUTTINGDOWN(wlxAction))
+                    break;
```

Comment:
Merge these two `if` together?
And they could be moved to before the previous `if`?